plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/TableStatisticsWriter.java

```diff
@@ -196,49 +195,81 @@ private CollectedStatistics mergeStatisticsIfNecessary(
         }
 
         return switch (updateMode) {
-            case REPLACE -> collectedStatistics;
+            case REPLACE -> {
+                Map<Integer, CompactSketch> ndvSketches = updateStats(
+                        collectedStatistics,
+                        table,
+                        snapshotId,
+                        fileIO,
+                        false);
+                yield new CollectedStatistics(ndvSketches);
+            }
             case INCREMENTAL_UPDATE -> {
-                Map<Integer, CompactSketch> collectedNdvSketches = collectedStatistics.ndvSketches();
-                ImmutableMap.Builder<Integer, CompactSketch> ndvSketches = ImmutableMap.builder();
+                Map<Integer, CompactSketch> ndvSketches = updateStats(
+                        collectedStatistics,
+                        table,
+                        snapshotId,
+                        fileIO,
+                        true);
+                yield new CollectedStatistics(ndvSketches);
+            }
+        };
+    }
 
-                Set<Integer> pendingPreviousNdvSketches = new HashSet<>(collectedNdvSketches.keySet());
-                Iterator<StatisticsFile> statisticsFiles = walkStatisticsFiles(table, snapshotId);
-                while (!pendingPreviousNdvSketches.isEmpty() && statisticsFiles.hasNext()) {
-                    StatisticsFile statisticsFile = statisticsFiles.next();
+    private Map<Integer, CompactSketch> updateStats(
+            CollectedStatistics collectedStatistics,
+            Table table,
+            long snapshotId,
+            FileIO fileIO,
+            boolean shouldMergeWithOld)
```

Comment:
is `shouldMergeWithOld` equivalent to `updateMode`?

you could inline the method, i.e. remove

```
 return switch (updateMode) {
            case REPLACE -> {
                Map<Integer, CompactSketch> ndvSketches = updateStats(
                        collectedStatistics,
                        table,
                        snapshotId,
                        fileIO,
                        false);
                yield new CollectedStatistics(ndvSketches);
            }
            case INCREMENTAL_UPDATE -> {
                Map<Integer, CompactSketch> ndvSketches = updateStats(
                        collectedStatistics,
                        table,
                        snapshotId,
                        fileIO,
                        true);
                yield new CollectedStatistics(ndvSketches);
            }
        };
    }

    private Map<Integer, CompactSketch> updateStats(
            CollectedStatistics collectedStatistics,
            Table table,
            long snapshotId,
            FileIO fileIO,
            boolean shouldMergeWithOld)
    {
```